exo/networking/manual/manual_discovery.py

```diff
@@ -42,17 +38,30 @@ async def discover_peers(self, wait_for_peers: int = 0) -> List[PeerHandle]:
     if DEBUG_DISCOVERY >= 2: print(f"Discovered peers: {[peer.id() for peer in self.known_peers.values()]}")
     return list(self.known_peers.values())
 
+  async def task_clean_up_peers_from_config(self):
+    if DEBUG_DISCOVERY >= 2: print("Starting task to clean up peers from config...")
+    while True:
+      peers_from_config = self._get_peers().items()
+      if peers_from_config:
+        peers_to_remove = [peer for peer in self.known_peers.keys() if peer not in peers_from_config]
+
+        for peer in peers_to_remove:
+          if DEBUG_DISCOVERY >= 2: print(f"{peer} is no longer found in the config. Removing...")
+          try: del self.known_peers[peer]
+          except KeyError: pass
+
+      await asyncio.sleep(5.0)
```

Comment:
I think a cleanup of "stale" peers (one that has been added to the know_peers list but is no longer in the config file) is now necessary.

Related to my other comment around handling the WIP state of the config file, only thing I'm unsure of it how long it should sleep for. I feel like the delay on looking for stale peers should be longer than how often we try and find/add new peers, but I'm not 100% sure.